### PR TITLE
Added a service to xor `Bytes`

### PR DESCRIPTION
**Pull request by ppepos**
Needed a new constructor for `Bytes` to facilitate things.

### CODE REVIEW

File: lib/core/bytes.nit [hunk 1]
```diff
@@ -173,6 +173,17 @@ class Bytes
 		init(ns, 0, cap)
 	end
 
+	# Init a `Bytes` from a string
+	init from(s: Text) do
```
Comment (privat) [inline lib/core/bytes.nit]: Why not using `Text::to_bytes`?

File: lib/crypto.nit [hunk 2]
```diff
@@ -183,6 +183,36 @@ redef class Text
 	end
 end
 
+redef class Bytes
+
+	# Returns `self` xored with `key`
+	#
+	# The shortest of the two is cycled through until the longest has been
+	# completely xored.
```
Comment (privat) [inline lib/crypto.nit]: I'm not sure if using the shortest is a POLA approach.
I assumed that xorcrypting was symmetric. ie `message.xor(key).xor(key) == message`. But if key.length > message.length this become false.

Comment (privat) [inline lib/crypto.nit]: also rename the method `xorcipher` or else people will mistake it with the mathematical bitwise xor operation (ie without cycling).

**Comment by ppepos** [comment]
* Removed useless constructor.
* Added https://github.com/nitlang/nit/pull/2061/commits/2580cc81efbb38638f26b0d6c16630e7a981e1f2
* Changed algorithm to not pick longest as `message` and shortest as `key` but to always consider `self`as `message`.

**Comment by privat** [comment]
+1

**Comment by jcbrinfo** [comment]
:+1: +1

**Pull request merged**